sdk/cmake/gcc.cmake

```diff
@@ -440,3 +415,47 @@ function(add_linker_script _target _linker_script_file)
     add_target_link_flags(${_target} "-Wl,-T,${_file_full_path}")
     add_target_property(${_target} LINK_DEPENDS ${_file_full_path})
 endfunction()
+
+# Manage our C++ options
+# we disable standard includes if we don't use the STL
+add_compile_options("$<$<AND:$<COMPILE_LANGUAGE:CXX>,$<NOT:$<IN_LIST:cppstl,$<TARGET_PROPERTY:LINK_LIBRARIES>>>>:-nostdinc>")
+# we disable RTTI, unless said so
+add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:$<IF:$<BOOL:$<TARGET_PROPERTY:WITH_CXX_RTTI>>,-frtti,-fno-rtti>>")
+# We disable exceptions, unless said so
+add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:$<IF:$<BOOL:$<TARGET_PROPERTY:WITH_CXX_EXCEPTIONS>>,-fexceptions,-fno-exceptions>>")
+
+# Find default G++ libraries
+add_library(libgcc STATIC IMPORTED)
+execute_process(COMMAND ${CMAKE_CXX_COMPILER} -print-file-name=libgcc.a OUTPUT_VARIABLE LIBGCC_LOCATION)
+string(STRIP ${LIBGCC_LOCATION} LIBGCC_LOCATION)
+set_target_properties(libgcc PROPERTIES IMPORTED_LOCATION ${LIBGCC_LOCATION})
+# libgcc needs kernel32 imports and msvcrtex
+target_link_libraries(libgcc INTERFACE libkernel32 msvcrtex)
```

Comment:
This looks like it is pulling in host libraries?
Do we want that?

Reply:
Strange that this block is now needed whereas it was not before....

Reply:
> This looks like it is pulling in host libraries?
> Do we want that?

That's what we already do through -lgcc, -lsupc++ and -lstdc++
And no, those are not host libraries, these are the libraries the GCC provides along its installation

Reply:
> Strange that this block is now needed whereas it was not before....

We link to this libraries through the -lgcc, -lsupc++ and -lstdc++ compiler flags. It's more explicit this way. This is needed because you can't have an imported static library without an actual file behind it, so you have to ask G++ for its location.

Now the target_link_libraries part is needed to have cmake properly handle the linking order, since now those are actual library for it, and not random linker flags